File: src/types/relationship.ts
export type RelationshipType =
  | 'friend'
  | 'colleague'
  | 'manager'
  | 'reports_to'
  | 'partner'
  | 'family'
  | 'mentor'
  | 'mentee'
  | 'client'
  | 'customer'
  | 'supplier'
  | 'teammate'
  | 'collaborator'
  | 'rival'
  | 'custom';

export type RelationshipStrength = 'weak' | 'normal' | 'strong';

export type RelationshipStatus = 'active' | 'paused' | 'ended';

export interface Relationship {
  id: string;
  worldId: string;
  fromPersonId: string;
  toPersonId: string;
  type: RelationshipType;
  strength: RelationshipStrength;
  status: RelationshipStatus;
  description?: string;
  createdAt: string;
  updatedAt: string;
}

File: backend/app/schemas/relationship.py
from pydantic import BaseModel, Field
from typing import Optional

class RelationshipBase(BaseModel):
    worldId: str
    fromPersonId: str
    toPersonId: str
    type: str = "colleague"
    strength: str = "normal"
    status: str = "active"
    description: Optional[str] = None

class RelationshipCreate(RelationshipBase):
    pass

class RelationshipUpdate(BaseModel):
    type: Optional[str] = None
    strength: Optional[str] = None
    status: Optional[str] = None
    description: Optional[str] = None

class RelationshipResponse(RelationshipBase):
    id: str
    createdAt: str
    updatedAt: str

    class Config:
        from_attributes = True
